Tanvir04Khan/01-max-udemy/03-login-page/src/Components/Login/LoginPage.js

```diff
@@ -0,0 +1,101 @@
+import React, { useState, useEffect, useReducer, Fragment } from "react";
+
+import Card from "../UI/Card";
+import UserInput from "./UserInput";
+import Header from "./Header";
+
+import "./LoginPage.css";
+
+const emailReducer = (state, action) => {
+  if (action.type === "email-input") {
+    return {
+      value: action.val,
+      isValid: action.val.includes("@") && action.val.includes(".com"),
+    };
+  }
+  return { value: "", isValid: false };
+};
+
+const passwordReducer = (state, action) => {
+  if (action.type === "password-input") {
+    return { value: action.val, isValid: action.val.trim().length >= 6 };
+  }
+  return { value: "", isValid: false };
+};
+
+const LoginPage = ({ isLoggedin }) => {
+  const [enteredEmail, dispatchEmail] = useReducer(emailReducer, {
+    value: "",
+    isValid: false,
+  });
+  const [enteredPassword, dispatchPassword] = useReducer(passwordReducer, {
+    value: "",
+    isValid: false,
+  });
+
+  const [formIsValid, setFormIsValid] = useState(false);
+
+  useEffect(() => {
+    const identifier = setTimeout(() => {
+      console.log("checking form validation!");
+      setFormIsValid(enteredEmail.isValid && enteredPassword.isValid);
+
+      return () => {
+        clearTimeout(identifier);
+      };
+    }, 1000);
+  }, [enteredEmail, enteredPassword]);
+
+  const emailInputHandler = (event) => {
+    dispatchEmail({ type: "email-input", val: event.target.value });
+  };
+
+  const passwordInputHandler = (event) => {
+    dispatchPassword({ type: "password-input", val: event.target.value });
+  };
+
+  const loginHandler = (event) => {
+    event.preventDefault();
+    if (formIsValid) {
+      isLoggedin(true);
+      localStorage.setItem("isLoggedIn", "1");
```

Comment:
why not set it to true? so that u wont need to compare it with a string 1 in App.js on line 12

Reply:
changed value as per suggestion @AlishaPrasad